podpac/core/data/datasource.py

```diff
@@ -368,42 +376,45 @@ def _eval(self, coordinates, output=None, _selector=None):
         # Check the coordinate_index_type
         if self.coordinate_index_type == "slice":  # Most restrictive
             new_rsci = []
-            for rsci in self._requested_source_coordinates_index:
-                if isinstance(rsci, slice):
-                    new_rsci.append(rsci)
+            for index in rsci:
+                if isinstance(index, slice):
+                    new_rsci.append(index)
                     continue
 
-                if len(rsci) > 1:
-                    mx, mn = np.max(rsci), np.min(rsci)
-                    df = np.diff(rsci)
+                if len(index) > 1:
+                    mx, mn = np.max(index), np.min(index)
+                    df = np.diff(index)
                     if np.all(df == df[0]):
                         step = df[0]
                     else:
                         step = 1
                     new_rsci.append(slice(mn, mx + 1, step))
                 else:
-                    new_rsci.append(slice(np.max(rsci), np.max(rsci) + 1))
+                    new_rsci.append(slice(np.max(index), np.max(index) + 1))
 
-            self._requested_source_coordinates_index = tuple(new_rsci)
+            rsci = tuple(new_rsci)
 
         # get data from data source
-        self._requested_source_data = self._get_data(
-            self._requested_source_coordinates, self._requested_source_coordinates_index
-        )
+        rsd = self._get_data(rsc, rsci)
```

Comment:
This used to take no inputs and use `self._requested_source_coordinates_index` etc. internally. I was never able to follow this code so I refactored it, and that's how I introduced the bug. Thanks for catching it +1

Reply:
Well, thanks for finally fixing it so that we don't those internal variables.